spec/controllers/exchanges/scheduled_events_controller_spec.rb

```diff
@@ -0,0 +1,125 @@
+require 'rails_helper'
+
+RSpec.describe Exchanges::ScheduledEventsController do
+
+  let(:user){ double(:save => double("user")) }
+  let(:person) { FactoryGirl.create(:person) }  
+  let(:event_params) { {
+      type: 'holiday',
+      event_name: 'Christmas',
+      offset_rule: 3,
+      recurring_rules: {},
+      :start_time => Date.today
+    }}
+  after do
+    ScheduledEvent.delete_all
+  end
+  before do
+    sign_in(user)
+  end
+  
+  describe "GET new" do
+	it "should render the new template" do
+	  get :new
+      expect(response).to have_http_status(:success)
+    end
+  end
+
+  describe "Create Post" do
+    let(:scheduled_event) { ScheduledEvent.new(event_params) }
+  	
+  	it "returns http status" do
+  	  scheduled_event.stub(:save).and_return true
+      post :create, scheduled_event: event_params
+      expect(response).to redirect_to exchanges_scheduled_events_path
+    end
+    it "should render new template when invalid params" do
+      scheduled_event.stub(:save).and_return false
+      post :create, scheduled_event: event_params
+      expect(response).to redirect_to exchanges_scheduled_events_path
+    end
+  end
+
+  describe "destroy" do
+    let(:scheduled_event) { FactoryGirl.create(:scheduled_event) }
+    before :each do
+      delete :destroy, id: scheduled_event.id
+    end
+    it "redirects_to index page" do
+      expect(response).to redirect_to exchanges_scheduled_events_path
+    end
+  end
+
+  describe "update" do
+    context "remove event exceptions" do
+      let!(:event_exception) { FactoryGirl.create(:event_exception) }
+      let!(:scheduled_event) { ScheduledEvent.all.first }
+      it "delete event exceptions" do
+        put :update, id: scheduled_event.id, scheduled_event: event_params
+        expect(ScheduledEvent.all.first.event_exceptions.length).to eq 0
+        expect(response).to redirect_to exchanges_scheduled_events_path
+      end
+    end
+    context "Add recurring rules" do
+      let!(:scheduled_event) { FactoryGirl.create(:scheduled_event) }
+      it "set one time false with recurring rules" do
+        scheduled_event.update_attributes!(recurring_rules: "{\"interval\":1,\"until\":null,\"count\":null,\"validations\":{\"day_of_week\":{},\"day_of_month\":[22]},\"rule_type\":\"IceCube::MonthlyRule\",\"week_start\":0}")
+        put :update, id: scheduled_event.id, scheduled_event: event_params
+        expect(scheduled_event.one_time).to eq true
+        expect(response).to redirect_to exchanges_scheduled_events_path
+      end
+    end
+  end
+
+  describe "GET index" do
+    let(:scheduled_event) { FactoryGirl.create(:scheduled_event) }
+    let(:scheduled_events) { [scheduled_event] }
+    it 'assigns scheduled_events' do
+      get :index, scheduled_event: event_params
+      assigns[:scheduled_events].should_not be_nil
+    end
+  end
+
+  describe "GET#get_system_events" do    
```

Comment:
Updated these in PR's https://github.com/health-connector/enroll/pull/88, https://github.com/health-connector/enroll/pull/86